controllers/component_controller.go

```diff
@@ -171,12 +171,14 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	log.Info(fmt.Sprintf("Starting reconcile loop for %v", req.NamespacedName))
 
 	if component.Spec.ContainerImage == "" {
-		component.Spec.ContainerImage = r.ImageRepository + ":" + component.Namespace + "-" + component.Name
+		userUniqueHash := util.GenerateUserUniqueHash(req.ClusterName, component.Namespace)
+		outputImage := r.ImageRepository + ":" + userUniqueHash + "-" + component.Name
+		component.Spec.ContainerImage = outputImage
```

Comment:
maybe just directly do `component.Spec.ContainerImage = r.ImageRepository + ":" + userUniqueHash + "-" + component.Name` so then the change will be minimalistic (no need to update log.Error and log.Info on following lines)

Reply:
Done